packages/backend/src/app/ee/pieces/platform-piece-module.ts

```diff
@@ -0,0 +1,54 @@
+import { FastifyPluginAsyncTypebox } from '@fastify/type-provider-typebox'
+import { FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox'
+import { ActivepiecesError, AddPieceRequestBody, ErrorCode, isNil } from '@activepieces/shared'
+import { PieceMetadataModel } from '../../pieces/piece-metadata-entity'
+import { platformService } from '../platform/platform.service'
+import { pieceService } from '../../pieces/piece-service'
+
+
+export const platformPieceModule: FastifyPluginAsyncTypebox = async (app) => {
+    await app.register(platformPieceController, { prefix: '/v1/pieces' })
+}
+
+const platformPieceController: FastifyPluginCallbackTypebox = (app, _opts, done) => {
+    app.post('/', {
+        schema: {
+            body: AddPieceRequestBody,
+        },
+    }, async (req): Promise<PieceMetadataModel> => {
+        const { platformId } = req.body
+        const { packageType, pieceName, pieceVersion, pieceArchive } = req.body
+        const { projectId } = req.principal
+        await assertUserIsPlatformOwner({
+            platformId,
+            userId: req.principal.id,
+        })
+        return await pieceService.add({
+            packageType,
+            pieceName,
+            pieceVersion,
+            archive: pieceArchive as Buffer,
+            projectId,
+        })
+
+    })
+
+    done()
+}
+
+async function assertUserIsPlatformOwner({
+    platformId,
+    userId,
+}: { platformId?: string, userId: string }): Promise<void> {
+    if (!isNil(platformId)) {
+        const platform = await platformService.getOneByOwner({
+            ownerId: userId,
+        })
+        if (platform?.id !== platformId) {
+            throw new ActivepiecesError({
+                code: ErrorCode.AUTHORIZATION,
+                params: {},
+            })
+        }
+    }
+}
```

Comment:
BTW, there's already a `checkUserIsOwner` method in the `platformService`!

(blocking)